streams/src/main/java/org/apache/kafka/streams/processor/assignment/TaskAssignmentUtils.java

```diff
@@ -407,4 +543,345 @@ private static boolean hasValidRackInformation(final TaskInfo task,
         }
         return true;
     }
+
+    private static Map<ProcessId, KafkaStreamsAssignment> tagBasedStandbyTaskAssignment(final ApplicationState applicationState,
+                                                                                        final Map<ProcessId, KafkaStreamsAssignment> kafkaStreamsAssignments) {
+        final int numStandbyReplicas = applicationState.assignmentConfigs().numStandbyReplicas();
+        final Map<TaskId, Integer> tasksToRemainingStandbys = applicationState.allTasks().values().stream()
+            .collect(Collectors.toMap(TaskInfo::id, taskInfo -> numStandbyReplicas));
```

Comment:
nit: this isn't a correctness issue, but for logging/debugging/inspection sanity we should only fill in this map with stateful tasks. Technically the "remaining standbys" for any stateless tasks is just 0 (though I'd rather just not include them at all in this map, not suggesting you set their value to 0)

We can also save a small bit of processing overhead as well by just building this map based on the `statefulTaskIds` set that's initialized a bit later. We can move this down to after that gets created and then simplify it to just `statefulTaskIds.stream().collect(toMap(t -> t, t -> numStandbyReplicas))`